api/wasm.go

```diff
@@ -162,6 +146,30 @@ type Module interface {
 
 	// ExportedGlobal a global exported from this module or nil if it wasn't.
 	ExportedGlobal(name string) Global
+
+	// ModuleCloser closes this module.
+	ModuleCloser
+}
+
+// ModuleCloser closes a module with an exit code.
+//
+// Note: This is an interface for decoupling, not third-party implementations. All implementations are in wazero.
+type ModuleCloser interface {
```

Comment:
hmm how about Closer? this also closes the runtime. shouldn't have any conflicts as `api.Closer` is different than say `io.Closer`